qiskit/qasm3/exporter.py

```diff
@@ -591,38 +583,25 @@ def build_opaque_definition(self, instruction):
             f"\n{instruction}"
         )
 
-    def build_subroutine_definition(self, instruction):
-        """Builds a SubroutineDefinition"""
-        if instruction.definition.parameters:
-            # We don't yet have the type system to store the parameter types in a symbol table, and
-            # we currently don't have the correct logic in place to handle parameters correctly in
-            # the definition.
-            raise QASM3ExporterError(
-                "Exporting subroutines with parameters is not yet supported by the OpenQASM 3"
-                " exporter. Received this instruction, which appears parameterized:"
-                f"\n{instruction}"
+    def build_gate_definition(self, gate):
+        """Builds a QuantumGateDefinition"""
+        if isinstance(gate, standard_gates.CXGate):
+            # CX gets super duper special treatment because it's the base of Terra's definition
+            # tree, but isn't an OQ3 built-in.  We use `isinstance` because we haven't fully
+            # fixed what the name/class distinction is (there's a test from the original OQ3
+            # exporter that tries a naming collision with 'cx').
```

Comment:
The same comment appears above. We probably don't need it twice. The comments are a bit mysterious. I'm not sure if they should be clarified, or maybe I just have to read more code to understand.

Reply:
Which bits of it would you like me to clarify?  I had it in both places because I thought they were sufficiently far apart in the code that it warranted it, but I'm happy to reduce them if you think it'd be clearer.

Reply:
Since putting the comments in both places was intentional, I'm ok with leaving them both in place.